src/TableOfContents/index.tsx

```diff
@@ -0,0 +1,226 @@
+
+  // TODO (CL): Should we store expanded state in local storage?
+  const [expanded, setExpanded] = React.useState({});
+
+  // Whenever the SRN changes, make sure the parent is expanded
+  // TODO (CL): Handle deeply nested expanding
+  React.useEffect(() => {
+    if (!srn) return;
+
+    // Get the index of the currently active item
+    const index = contents.findIndex(item => item.isActive);
+    if (index === -1 || !contents[index] || contents[index].depth === 0) return;
+
+    const parentIndex = findParentIndex(contents[index].depth, contents.slice(0, index));
+    if (parentIndex > -1 && !expanded[parentIndex]) {
+      setExpanded({ ...expanded, [parentIndex]: true });
+    }
+  }, [srn]);
+
+  const isMobile = useIsMobile(enableDrawer);
+
+  const comp = (
+    <div className={cn('TableOfContents bg-gray-1 dark:bg-transparent flex justify-end h-full', className)}>
+      <div className="w-full">
+        <ScrollContainer>
+          <div className={cn('TableOfContents__inner ml-auto', `py-${padding}`)}>
+            {contents.map((item, index) => {
+              if (item.depth > 0) {
+                // Check if we should show this item
+                const parentIndex = findParentIndex(item.depth, contents.slice(0, index));
+                if (parentIndex > -1 && !expanded[parentIndex]) {
+                  return null;
+                }
+              }
+
+              const isGroup = item.type === 'group';
+              const isDivider = item.type === 'divider';
+              const isExpanded = expanded[index];
+
+              return (
+                <TableOfContentsItem
+                  key={index}
+                  name={item.name}
+                  // srn={item.srn}
+                  depth={item.depth}
+                  isActive={item.isActive || false}
+                  isGroup={isGroup}
+                  isExpanded={isExpanded}
+                  isDivider={isDivider}
+                  onClick={e => {
+                    if (isDivider) {
+                      e.preventDefault();
+                      return;
+                    }
+
+                    if (!isGroup) return;
+
+                    e.preventDefault();
+                    setExpanded({ ...expanded, [String(index)]: !isExpanded });
+                  }}
+                  icon={item.icon}
+                />
+              );
+            })}
+          </div>
+        </ScrollContainer>
+      </div>
+    </div>
+  );
+
+  if (isMobile) {
+    const { org, project } = deserializeSrn(srn || '');
+    return (
+      <Drawer isOpen={isOpen} onClose={() => onClose()} position="left" size="330px">
+        <div className="flex flex-1 flex-col bg-gray-1 dark:bg-transparent">
+          <div className="border-b dark:border-lighten-4 h-20 py-6 px-2 bg-white">
+            <Button className="flex justify-start text-lg" icon={'arrow-left'} minimal onClick={() => onClose()}>
+              {title || `${org} / ${project}`}
```

Comment:
We won't have "org" and "project" anymore since we're getting rid of SRN. Elements should pass in the correct title though

```
export const TableOfContents = ({ items, srn, title }) => {
  const contents = useComputeToc(items);
  const { org, project } = deserializeSrn(srn);

  return (
    <UIKitTableOfContents
      contents={contents}
      title={title || `${org} / ${project}`}
    />
  );
};
```